src/migrations/tasks/20200126113825_cwu_init.ts

```diff
@@ -0,0 +1,163 @@
+    table.uuid('opportunityVersion').references('id').inTable('cwuOpportunityVersions').notNullable();
+    table.uuid('file').references('id').inTable('files').notNullable();
+  });
+  logger.info('Created cwuOpportunityAttachments table.');
+
+  // Create CWUOpportunityAddendum table
+  await connection.schema.createTable('cwuOpportunityAddenda', table => {
+    table.uuid('id').primary().unique().notNullable();
+    table.uuid('opportunity').references('id').inTable('cwuOpportunities').notNullable();
+    table.timestamp('createdAt').notNullable();
+    table.uuid('createdBy').references('id').inTable('users').notNullable();
+    table.string('description').notNullable();
+  });
+  logger.info('Created cwuOpportunityAddenda table.');
+
+  // Create CWUProposal table
+  await connection.schema.createTable('cwuProposals', table => {
+    table.uuid('id').primary().unique().notNullable();
+    table.timestamp('createdAt').notNullable();
+    table.uuid('createdBy').references('id').inTable('users').notNullable();
+    table.timestamp('updatedAt').notNullable();
+    table.uuid('updatedBy').references('id').inTable('users').notNullable();
+    table.string('proposalText');
+    table.string('additionalComments');
+    table.uuid('proponentOrg').references('id').inTable('organizations');
+    table.float('score');
+  });
+  logger.info('Created cwuProposals table.');
+
+  // Create CWUProponent table
+  await connection.schema.createTable('cwuProponents', table => {
+    table.uuid('id').primary().unique().notNullable();
+    table.timestamp('createdAt').notNullable();
+    table.uuid('createdBy').references('id').inTable('users').notNullable();
+    table.timestamp('updatedAt').notNullable();
+    table.uuid('updatedBy').references('id').inTable('users').notNullable();
+    table.string('legalName').notNullable();
+    table.string('email').notNullable();
+    table.string('phone');
+    table.string('streetAddress1').notNullable();
+    table.string('streetAddress2');
+    table.string('city').notNullable();
+    table.string('region').notNullable();
+    table.string('mailCode').notNullable();
+    table.string('country').notNullable();
+  });
+  logger.info('Created cwuProponents table.');
+
+  // Add foreign key relationship Proposals --> Proponents
+  await connection.schema.alterTable('cwuProposals', table => {
+    table.uuid('proponent').references('id').inTable('cwuProponents');
+  });
+  logger.info('Foreign key \'proponent\' added to cwuProposals table.');
+
+  // Add constraint to CWUProposal table (proponent is either individual or org - not both)
+  await connection.schema.raw(`
+  ALTER TABLE "cwuProposals"
+  ADD CONSTRAINT "eitherProponentOrOrg" check(
+    ("proponent" IS NOT NULL AND "proponentOrg" IS NULL)
+    OR
+    ("proponent" IS NULL AND "proponentOrg" IS NOT NULL)
+  )
+  `);
+  logger.info('Added constraint \'eitherProponentOrOrg\' to cwuProposals table.');
```

Comment:
👍 
Not super picky about this, but you could rename the `proponent` column to something like `proponentIndividual` for added clarity

Reply:
Good idea, done.